### PR TITLE
Exclude code from SonarCloud analysis

### PR DESCRIPTION
**Pull request by tpantelis**
- Generated code
- Test support
- External library wrappers

### CODE REVIEW

**Review by skitt — APPROVED** [review]

File: sonar-project.properties [hunk 1]
```diff
@@ -2,7 +2,7 @@ sonar.projectKey=submariner-io_submariner
 sonar.projectName=submariner
 sonar.organization=submariner-io
 sonar.sources=.
-sonar.exclusions=**/vendor/**,**/*_test.go,**/test/**,**/fake/**,**/client/**
+sonar.exclusions=**/vendor/**,**/*_test.go,**/test/**,**/fake/**,**/client/**,**/natdiscovery/proto/natdiscovery.pb.go,**/netlink/**,**/event/testing/**,**/zz_generated.deepcopy.go,**/ipset/**
```
Comment (sridhargaddam) [inline sonar-project.properties:5]: Is the removal of ipset and netlink packages intentional?

Comment (skitt) [reply sonar-project.properties:5]: IIUC, yes, because they are external library wrappers.

Comment (sridhargaddam) [reply sonar-project.properties:5]: Not all APIs are passthrough, we have some APIs implemented. Do we want to avoid sonar on them?

Comment (tpantelis) [reply sonar-project.properties:5]: `netlink` is a straight pass-through. `ipset` is not our code - we copied it from somewhere. There might be some helper methods we could unit test but I don't think it's necessary. I think the E2E tests it sufficiently, at least the code we use. @sridhargaddam What API are you specifically concerned about?

Comment (tpantelis) [reply sonar-project.properties:5]: Actually, FWICT, virtually all helpers, eg validate*, are private.

Comment (sridhargaddam) [reply sonar-project.properties:5]: Right, there are few APIs like Create/Add/Del/Destroy etc, but the code is pretty much stable and has not undergone any major changes, so we can skip it. I wanted to confirm that we did not accidentally remove them.

**Review by sridhargaddam — APPROVED** [review]

**Pull request merged**